test/unit/overlay.js

```diff
@@ -6,19 +6,78 @@ var sharp = require('../../index');
 
 sharp.cache(0);
 
-// Main
+
+// Constants
+var MAX_ALLOWED_IMAGE_MAGICK_MEAN_SQUARED_ERROR = 0.3;
+
+// Helpers
+var getPaths = function(baseName, extension) {
+  if (typeof extension === 'undefined') {
+    extension = 'png';
+  }
+
+  var actual = fixtures.path('output.' + baseName + '.' + extension);
+  var expected = fixtures.expected(baseName + '.' + extension);
+  var expectedMagick = fixtures.expected(baseName + '-imagemagick.' + extension);
```

Comment:
Since I already generated and checked in ImageMagick reference images, I use them in tests here.